src/owee_buf.ml

```diff
@@ -69,14 +71,18 @@ module Read = struct
   let u32be = u32
 
   let u64 t : u64 =
-    let result = t.buffer.{t.position}
-                 lor t.buffer.{t.position + 1} lsl 8
-                 lor t.buffer.{t.position + 2} lsl 16
-                 lor t.buffer.{t.position + 3} lsl 24
-                 lor t.buffer.{t.position + 4} lsl 32
-                 lor t.buffer.{t.position + 5} lsl 40
-                 lor t.buffer.{t.position + 6} lsl 48
-                 lor t.buffer.{t.position + 7} lsl 56
+    let result = List.fold_left
+                   (fun acc n -> Int64.logor acc (Int64.of_int n))
+                   0L
+                   [ t.buffer.{t.position};
+                     t.buffer.{t.position + 1} lsl 8;
+                     t.buffer.{t.position + 2} lsl 16;
+                     t.buffer.{t.position + 3} lsl 24;
+                     t.buffer.{t.position + 4} lsl 32;
+                     t.buffer.{t.position + 5} lsl 40;
+                     t.buffer.{t.position + 6} lsl 48;
+                     t.buffer.{t.position + 7} lsl 56
+                   ]
```

Comment:
Oops, this is wrong, you are shifting in the `int` domain, that defeats the purpose of switching to `int64` :).

Two alternatives implementation:
```
type buf = (int, Bigarray.int8_unsigned_elt, Bigarray.c_layout) Bigarray.Array1.t

let read_fun (b : buf) p =
  let b i acc = Int64.(logor (shift_left (of_int b.{p + i}) (i * 8)) acc) in
  0L |> b 0 |> b 1 |> b 2 |> b 3 |> b 4 |> b 5 |> b 6 |> b 7

let read_imp (b : buf) p =
  let acc = ref 0L in
  for i = 0 to 7 do
    acc := Int64.(logor !acc (shift_left (of_int b.{p + i}) (i * 8)))
  done;
  !acc
```

Reply:
oh no :( 
I've just applied your fix using imperative version, thanks!!